### PR TITLE
SSL improvements and profile settings

### PR DESCRIPTION
**Pull request by ghost**
Changed generating of SSL certificates for better compatibility with chrome browser.

Self-Signed certs get now added to trusted on vm, especially nice for multi-host-systems which need to communicate secure.

generated ssl certificates get copied to vagrant-php/certs for easy importing on host system.

Added a profile role for profile related settings, added an alias "ll"

looking forward for feedback.

Best, Sam

### CODE REVIEW

File: ansible/roles/nginx/files/openssl.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# Call this script with at least 10 parameters, for example
```
Comment (plandolt) [inline ansible/roles/nginx/files/openssl.sh]: 10 -> 1

File: ansible/playbook.yml [hunk 2]
```diff
@@ -65,3 +65,4 @@
     - { role: redis, when: role.redis }
     - { role: zsh, when: general.config.shell == 'zsh' }
     - cleanup
+    - profile
```
Comment (plandolt) [inline ansible/playbook.yml]: no need for that (see comment below)

File: ansible/roles/profile/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+- name: Add ll alias to vagrant user profile
+  lineinfile:
+    path: /home/vagrant/.profile
+    line: "alias ll='ls -la --color=auto'"
```
Comment (plandolt) [inline ansible/roles/profile/tasks/main.yml]: not needed. can be achieved with the configuration

```
bash:
    config:
        alias:
```

File: ansible/roles/nginx/files/openssl.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+if [[ ! -d "$EXPORT_CERT_DIR" ]]
+then
+  mkdir -p ${EXPORT_CERT_DIR};
+fi
+
+if [[ $# -eq 1 ]]
+then
+  echo
+  openssl req -nodes -x509 -newkey rsa:4096 -keyout /etc/nginx/$1.key -reqexts SAN -extensions SAN -out /etc/nginx/$1.crt -days 3650 -subj "/C=CH/ST=Zurich/L=Zurich/CN=$1" -config <(cat /etc/ssl/openssl.cnf; printf "[SAN]\nsubjectAltName=DNS:$1")
```
Comment (plandolt) [inline ansible/roles/nginx/files/openssl.sh]: `rsa:4096` should be changed to `rsa:2048` which is suitable enough for development

File: ansible/roles/nginx/files/openssl.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  cp -f /etc/nginx/$1.crt ${EXPORT_CERT_DIR}/.
+  cp -f /etc/nginx/$1.crt ${IMPORT_CERT_DIR}/.
+  update-ca-certificates
+  exit 0
+fi
+
+if [[ $# -eq 2 ]]
+then
+  echo
+  openssl req -nodes -x509 -newkey rsa:4096 -keyout /etc/nginx/$2.$1.key -reqexts SAN -extensions SAN -out /etc/nginx/$2.$1.crt -days 3650 -subj "/C=CH/ST=Zurich/L=Zurich/CN=$1" -config <(cat /etc/ssl/openssl.cnf; printf "[SAN]\nsubjectAltName=DNS:$1,DNS:$2.$1")
```
Comment (plandolt) [inline ansible/roles/nginx/files/openssl.sh]: `rsa:4096` should be changed to `rsa:2048` which is suitable enough for development

**Review by plandolt — CHANGES_REQUESTED** [review]

File: ansible/roles/nginx/files/openssl.sh (hunk 1 above)
Comment (ghost) [reply ansible/roles/nginx/files/openssl.sh]: fixed

File: ansible/playbook.yml (hunk 2 above)
Comment (ghost) [reply ansible/playbook.yml]: okay, fixed

File: ansible/roles/nginx/files/openssl.sh (hunk 5 above)
Comment (ghost) [reply ansible/roles/nginx/files/openssl.sh]: okay, fixed

**Review by plandolt — APPROVED** [review]

**Comment by plandolt** [comment]
ping @dominikzogg what do you thing about this PR? Or do you want to go with something like mkcert and load the locally generated certs into the vagrant box?

**Comment by dominikzogg** [comment]
@brsart i like the idea, @scuben do you think this will break existing projects if the update the submodule?

**Comment by plandolt** [comment]
Good question... @brsart please test it with a multihost setup the old way then updated to this PR and provision again. Also test multiple provisions with this PR.

**Comment by ghost** [comment]
Hi @dominikzogg and @scuben 

I have tested the updatability and figured out it runs all fine.

I am looking forward to your reply.

Best Sam


Test procedure multihost environment with 2 hosts:
1. Installed vagrant-php on original state:
    - Environment works successful

2. Second provisioning:
    - Environment works successful

3. Updated to newest commit of PR and run provisioning:
    - Environment works successful
    - Certificates created in “certs” folder and are valid

4. Reloaded vagrant with provisioning:
    - Environment works successful
    - Certificates re-created in “certs” folder and are valid
    - Needed to add the new certificates to trusted key-chain

**Comment by plandolt** [comment]
Do we want to reissue the certificates on reprovisioning?

**Comment by ghost** [comment]
We need kind a reissue of the certificates cause the can run out of validity, but we could do this e.g. only if they were removed from certs folder.

Looking forward to your suggestions.

**Comment by dominikzogg** [comment]
@brsart if i get this right, the feature is mergeable? with regeneration if the certs get deleted?

**Comment by ghost** [comment]
@dominikzogg i just figured out that we possibly have an issue with single host systems.
Please let me check and fix this issue, i'll reply as soon as possible.

**Comment by ghost** [comment]
@dominikzogg I've tested again and it's all fine, ready for merge.
Best, Samuel

**Comment by dominikzogg** [comment]
@brsart thanks for the effort, would be cool, if you would rebase to 1 commit. Would be easier to adapt to the 18.04 branch

**Comment by ghost** [comment]
Hallo @dominikzogg 

Habe die changes auf einen commit rebased :-)

Beste Grüsse
Samuel

**Comment by dominikzogg** [comment]
@brsart thanks for the contribution and @scuben for the support

**Pull request merged**